R/single_value.R

```diff
@@ -0,0 +1,29 @@
+#' Ensure that a vector has only a single value throughout.
+#'
+#' Missing values are replaced with the single value, and if all values are
+#' missing, the first value in \code{missing} is used throughout.
+#'
+#' @param x The vector which should have a single value
+#' @param missing The vector of values to consider missing in \code{x}
+#' @param warn_if_all_missing Generate a warning if all values are missing?
+#' @param info If more than one value is found, append this to the warning or
+#'   error to assist with determining the location of the issue.
+#' @return \code{x} as the scalar single value found throughout (or an error if
+#'   more than one value is found).
+#' @examples
+#' single_value(c(NA, 1))
```

Comment:
What do you think about this for `@examples` to show a typical use case?  Like what you showed in the original issue for this function.
```
library(dplyr)
# imputes a value where missing:
dat %>%
  dplyr::group_by(id) %>%
  mutate(age = single_value(age))

# fails because of multiple values:
\dontrun{
dat %>%
  dplyr::group_by(id) %>%
  mutate(country = single_value(country))
}
```

Reply:
I added a similar example.